src/main/java/io/javalin/core/compression/DynamicCompressionHandler.kt

```diff
@@ -0,0 +1,62 @@
+package io.javalin.core.compression
+
+import io.javalin.core.JavalinConfig
+import io.javalin.core.util.Header
+import io.javalin.http.Context
+import javax.servlet.http.HttpServletResponse
+
+/**
+ * Class designed to handle dynamic content compression for Javalin.
+ *
+ * Uses DynamicCompressionStrategy within JavalinConfig to determine the type and level of compression to be applied.
+ *
+ * @see DynamicCompressionStrategy
+ */
+class DynamicCompressionHandler(val ctx: Context, val config: JavalinConfig) {
+
+    private val compressionStrategy = config.inner.dynamicCompressionStrategy
+
+    /**
+     * @param res The response that we wish to encode
+     */
+    fun compressResponse(res: HttpServletResponse) {
+        val resultStream = ctx.resultStream()!!
+        if (brotliShouldBeDone(ctx)) { //Do Brotli
+            val level = compressionStrategy.brotliLevel
+            BrotliWrapper(level).use { brWrapper ->
```

Comment:
I guess I feel like `BrotliWrapper` (and `GzipWrapper`) should be utility methods rather than objects, but I haven't looked at it that closely yet. Why did you choose to make them objects? I was thinking something like:

```
BrotliWrapper.write(res)
```

Reply:
I made them objects to keep their usage consistent with Java's GZIPOutputStream that was originally being used.

I don't mind changing them to utility methods if you feel that is better. :)

Reply:
I just had a potentially great idea regarding these wrapper objects.

You know when you asked me to change the compression strategy constructor to something along the lines of "DynamicCompressionStrategy(Brotli(4), Gzip(6))"?

What if, instead of having separate data classes, we have DynamicCompressionStrategy initialize (and keep references to) the actual gzip/brotli wrappers instead?

So actual compression calls would then look something like this:

```
if(brotliShouldBeDone) {
    dynamicCompressionStrategy.brotli.write(res)
}
```

This would: 
- Avoid instantiating new objects for every compression op
- Eliminate data redundancy (compression levels being kept in two places)
- Remove the need to have separate data classes just to have a good looking strategy constructor
- brotli and gzip objects would only be initialized once with compression level params, and would then call static "write" methods. This is in line with the utility method approach that you envisioned

What do you think?

Reply:
Sounds like a very good approach!

Reply:
See my newest commit: 369adf6

I am still instantiating new GzipWrapper objects inside the Gzip class for every write, but that's because I couldn't find any way around it without over-complicating things, since the base Java class I am inheriting from (GZIPOutputStream) is designed to be used this way.

I have however hidden it within the Gzip object, at least, so it's something that happens automatically in the background when doing a gzip.write() operation.

Brotli, I was able to move away from new object creation entirely, because the base jbrotli.BrotliCompressor class is designed to be used as a helper object.